components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/GeckoLoginStorageDelegate.kt

```diff
@@ -78,68 +71,26 @@ class GeckoLoginStorageDelegate(
 
     @Synchronized
     override fun onLoginSave(login: Login) {
+        val validationDelegate = DefaultLoginValidationDelegate(storage = loginStorage)
         scope.launch {
             val existingLogin = login.guid?.let { loginStorage.get(it) }
-
-            when (getPersistenceOperation(login, existingLogin)) {
-                Operation.UPDATE -> {
-                    existingLogin?.let { loginStorage.update(it.mergeWithLogin(login)) }
+                ?: loginStorage.getStoredLoginByImportantFields(login)
+            when (validationDelegate.getNewLoginPersistenceAsync(login).await()) {
+                LoginValidationDelegate.Result.CanBeUpdated -> {
+                    val mergedLogin = requireNotNull(existingLogin).mergeWithLogin(login)
```

Comment:
It doesn't seem like `existingLogin` is guaranteed to be nonnull here.  Could we noop in that case instead of crashing?  It's what we're doing here with other errors anyway.